gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineErrorProne.java

```diff
@@ -250,10 +254,37 @@ private static void configureErrorProneOptions(
                         check);
                 return false;
             }
-            return true;
+            return filterOutPreconditions.test(check);
         });
     }
 
+    /** Filters out preconditions checks if the required libraries are not on the classpath. */
+    public static Predicate<String> filterOutPreconditions(FileCollection compileClasspath) {
+        boolean hasSafeLogging = !compileClasspath.filter(file -> file.getName().startsWith("safe-logging-")).isEmpty();
```

Comment:
can we not get the configuration (containing raw dependencies) instead of having to work on files?

Reply:
We can guess it but I couldn't find a direct way to extract this via the API if we start with the JavaCompile task

Reply:
nvm, I could do a pass through the source sets and find the one matching the `compileJava` task name